src/tests/class_tests/openms/source/Isotope_test.cpp

```diff
@@ -0,0 +1,165 @@
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+// 
+// --------------------------------------------------------------------------
+// $Maintainer: Timo Sachsenberg $
+// $Authors: Andreas Bertsch $
+// --------------------------------------------------------------------------
+//
+
+#include <OpenMS/CONCEPT/ClassTest.h>
+#include <OpenMS/test_config.h>
+
+///////////////////////////
+
+#include <OpenMS/CHEMISTRY/Isotope.h>
+#include <OpenMS/CHEMISTRY/ElementDB.h>
+
+using namespace OpenMS;
+using namespace std;
+
+///////////////////////////
+
+START_TEST(Isotope, "$Id$")
+
+/////////////////////////////////////////////////////////////
+
+Isotope* e_ptr = nullptr;
+Isotope* e_nullPointer = nullptr;
+START_SECTION(Isotope())
+	e_ptr = new Isotope;
+	TEST_NOT_EQUAL(e_ptr, e_nullPointer)
+END_SECTION
+
+START_SECTION(~Isotope())
+	delete e_ptr;
+END_SECTION
+
+IsotopeDistribution dist;
+string name("Name"), symbol("Symbol");
+unsigned int atomic_number(15);
+double mono_weight(0.123456789);
+
+e_ptr = nullptr;
+START_SECTION((Isotope(const std::string & name,
+            const std::string & symbol,
+            unsigned int atomic_number,
+            unsigned int neutrons,
+            double mono_weight,
+            double abundance,
+            double half_life,
+            Isotope::DecayMode dm)))
+	e_ptr = new Isotope(name, symbol, atomic_number, 10u, mono_weight, 0.6, 42, Isotope::DecayMode::UNKNOWN);	
+	TEST_NOT_EQUAL(e_ptr, e_nullPointer)
+END_SECTION
+
+START_SECTION( const Element* getElement() const)
+  const Element* el = e_ptr->getElement();
+  TEST_NOT_EQUAL(el, e_nullPointer)
+  TEST_EQUAL(el->getSymbol(), "P")
+END_SECTION
+
+START_SECTION(Isotope(const Isotope& Isotope))
+	Isotope copy(*e_ptr);
+	TEST_EQUAL(*e_ptr == copy, true)
+END_SECTION
+
+delete e_ptr;
+e_ptr = new Isotope;
+
+START_SECTION(void setHalfLife(double hl))
+	e_ptr->setHalfLife(8.6);
+	NOT_TESTABLE
+END_SECTION
+
+START_SECTION(double getHalfLife() const)
+	TEST_REAL_SIMILAR(e_ptr->getHalfLife(), 8.6)
+END_SECTION
+
+START_SECTION(void setAbundance(double hl))
+	e_ptr->setAbundance(0.6);
+	NOT_TESTABLE
+END_SECTION
+
+START_SECTION(double getAbundance() const)
+	TEST_REAL_SIMILAR(e_ptr->getAbundance(), 0.6)
+END_SECTION
+
+START_SECTION(void setNeutrons(int hl))
+	e_ptr->setNeutrons(10);
+	NOT_TESTABLE
+END_SECTION
+
+START_SECTION(int getNeutrons() const)
+	TEST_EQUAL(e_ptr->getNeutrons(), 10)
+END_SECTION
+
+START_SECTION(void setDecayMode(int hl))
+	e_ptr->setDecayMode(Isotope::DecayMode::ALPHA);
+	NOT_TESTABLE
+END_SECTION
+
+START_SECTION(int getDecayMode() const)
+	TEST_EQUAL(e_ptr->getDecayMode(), Isotope::DecayMode::ALPHA)
+END_SECTION
+
+START_SECTION( virtual bool isIsotope() )
+  TEST_EQUAL(e_ptr->isIsotope(), true)
+END_SECTION
+
+START_SECTION(bool isStable() const)
+  TEST_EQUAL(e_ptr->isStable(), false)
+END_SECTION
+
+START_SECTION(Isotope& operator = (const Isotope& Isotope))
+	Isotope e = *e_ptr;
```

Comment:
you are not testing `operator=` here. You are testing the Ctor.
Use
```
 e = *e_ptr;
```
on an existing `e`